aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -734,5 +736,34 @@ Tensor tensor_cuda(ArrayRef<T> values, const TensorOptions& options) {
   }
 AT_FORALL_SCALAR_TYPES_EXCEPT_HALF(TENSOR)
 #undef TENSOR
+
+Tensor from_file(std::string filename, c10::optional<bool> shared, c10::optional<long> size, const TensorOptions& options) {
+    AT_CHECK(!options.pinned_memory(), "tensors constructed from a file cannot be pinned");
+    size_t my_size = 0;
```

Comment:
Oh interesting. Is `size` really optional in this case? It looks to me that `size = 0` and `size = nullopt` are indistinguishable.

Actually, what I half expected to see was that the `nullopt` case would infer the number of elements based on the existing file size.

Reply:
Oh, I see that you do set the size to `actual_size` if you do this. But you're testing if `my_size = 0` which is fishy; if I said `size = 0`, surely I really meant to give me a zero size view on the file? The docs don't disambiguate this case clearly, but a strict reading suggests that you should end up with a zero element tensor.